config/interpolate_walk.go

```diff
@@ -85,7 +89,13 @@ func (w *interpolationWalker) Map(m reflect.Value) error {
 func (w *interpolationWalker) MapElem(m, k, v reflect.Value) error {
 	w.csData = k
 	w.csKey = append(w.csKey, k)
-	w.key = append(w.key, k.String())
+
+	if w.sliceIndex != -1 {
+		w.key = append(w.key, fmt.Sprintf("%d.%s", w.sliceIndex, k.String()))
+	} else {
+		w.key = append(w.key, k.String())
+	}
```

Comment:
I read this as "If we are in a slice, the key for this MapElem needs to be prefixed with the slice index."

Seems sort of weird that only MapElems get this treatment - I'd expect the SliceElem visit to do this work. Is there a reason that MapElems need to be special cased that I am missing?

Related: it'd be really great if we could wrap a unit test around this, not sure how much of a nightmare that would be though.